src/address.rs

```diff
@@ -83,15 +83,26 @@ impl<A> Address<A, Strong> {
     /// an actor will not be prevented from being dropped if only weak sinks, channels, and
     /// addresses exist.
     pub fn downgrade(&self) -> WeakAddress<A> {
-        Address(self.0.downgrade())
+        Address(self.0.to_tx_weak())
     }
 }
 
 /// Functions which apply only to addresses which can either be strong or weak.
 impl<A> Address<A, Either> {
     /// Converts this address into a weak address.
     pub fn downgrade(&self) -> WeakAddress<A> {
-        Address(self.0.downgrade())
+        Address(self.0.to_tx_weak())
+    }
+}
+
+/// Functions which apply to any kind of address, be they strong or weak.
+impl<A, Rc> Address<A, Rc>
+where
+    Rc: RefCounter + Into<Either>,
```

Comment:
Not sure if it's worth encapsulating this bound into a new RC trait that we actually export or not - it provides the same experience as before but also there isn't much benefit outside of that. Maybe a note on the doc of RefCounter would be enough

Reply:
I don't think it is worth it. Unless users call actually want to call `as_either`, the `Into` bound is not actually mandatory for the others. Perhaps we should demonstrate that in the public API test.

Reply:
Added a new test to the public API module.